catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaTopicPropertiesMetadata.java

```diff
@@ -6,12 +6,41 @@
 
 import com.datastrato.gravitino.connector.BasePropertiesMetadata;
 import com.datastrato.gravitino.connector.PropertyEntry;
-import java.util.Collections;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
+import java.util.List;
 import java.util.Map;
 
 public class KafkaTopicPropertiesMetadata extends BasePropertiesMetadata {
+  public static final String PARTITION_COUNT = "partition-count";
+  public static final String REPLICATION_FACTOR = "replication-factor";
+
+  private static final Map<String, PropertyEntry<?>> PROPERTIES_METADATA;
+
+  static {
+    List<PropertyEntry<?>> propertyEntries =
+        ImmutableList.of(
+            PropertyEntry.integerOptionalPropertyEntry(
+                PARTITION_COUNT,
+                "The number of partitions for the topic, if not specified, "
+                    + "will use the num.partition property in the broker",
+                true /* immutable */,
+                null /* default value */,
+                false /* hidden */),
+            // TODO: make REPLICATION_FACTOR mutable if needed
+            PropertyEntry.shortOptionalPropertyEntry(
```

Comment:
Why are we not using PropertyEntry.integerOptionalPropertyEntry?

Reply:
Because the REPLICATION_FACTOR parameter type in Kafka API is also short

Reply:
I think using `integerOptionalPropertyEntry` is enough.  No need to add new one.

Reply:
@jerryshao What's your suggestion?

Reply:
My feeling is that an integer is enough, not necessary to distinguish between integer and short.

Reply:
Here is the usage of the `REPLICATION_FACTOR` in admin client:
https://github.com/datastrato/gravitino/blob/4bb2ca291598058cfe77aed72d59500770df27f9/catalogs/catalog-messaging-kafka/src/main/java/com/datastrato/gravitino/catalog/kafka/KafkaCatalogOperations.java#L454-L457

If an integer is used, when a user provides a number beyond the short range, it may lead to unpredictable outcomes.
However, if a short is used, explicit exceptions can be thrown during parameter validation to alert the user.

If you think the above issue acceptable, I can proceed with changing it to an integer here.

Reply:
It is not a big deal, `short` is also fine.